### DISCUSSION TITLE
Does bazel take the BUILD file itself into account with dependency management?

### DISCUSSION THREAD

**Discussion by jason-sachs** [bazelbuild/bazel#21751] [Q&A]
Is the BUILD file itself included in the dependency graph?

If I have a rule R that output X depends on inputs A, B, C and tool T with command-line options Topt that reference some other library L, then the set of things that can cause changes in X include changes to any of the following:

- inputs A,B,C
- library L
- tool T
- options Topt
- rule R

I'm just trying to understand what would happen if we change the BUILD file itself. My expectation is that this may produce a different output X.

**Comment by moroten** [comment]
Yes, Bazel will analyse the content of the BUILD.bazel file again when it changes and potentially produce different actions (commands with inputs and outputs). The actions will also rerun if the command line, env, input file or output path changes.

For remote execution, the lookup key for an action is a digest (default SHA256) of the serialised action description. This means that your cache lookup will depend on the full action description, otherwise the remote execution system would not have all the data to execute the action. Bazel's internal caching has a different digest, but it still takes the whole action description into account.
